From cff631cb39b1b9d66b791fb51a6f2c3db8e60917 Mon Sep 17 00:00:00 2001
From: Sebastian Keller <skeller@gnome.org>
Date: Sat, 14 Jan 2023 17:48:33 +0100
Subject: [PATCH] clutter/text: Don't call clutter_text_set_buffer() on finalize

clutter_text_set_buffer() tries to freeze/thaw notify, which is not
allowed during finalize and recent glib versions started warning about
this. This call can simply be removed, because the buffer is already set
to NULL on dispose, making the call in finalize redundant.

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/2566
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2790>
---
 clutter/clutter/clutter-text.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/clutter/clutter/clutter-text.c b/clutter/clutter/clutter-text.c
index a294c4b73..ded041bd9 100644
--- a/clutter/clutter/clutter-text.c
+++ b/clutter/clutter/clutter-text.c
@@ -1826,7 +1826,6 @@ clutter_text_finalize (GObject *gobject)
 
   clutter_text_free_paint_volume (self);
 
-  clutter_text_set_buffer (self, NULL);
   g_free (priv->font_name);
 
   g_clear_object (&priv->input_focus);
-- 
